lib/shopify_api/rest/pagination_request.ex

```diff
@@ -0,0 +1,49 @@
+defmodule ShopifyAPI.REST.PaginatedRequest do
```

Comment:
Couldn't all of this live in REST.Request?

Currently it
- duplicates all the header/url creation/auth logic
- ignores the Shopify API version set in configuration
- ignores json decoding set in configuration

Reply:
Yes! Thanks, that makes a lot more sense.